ci/scripts/check-generated.sh

```diff
@@ -49,6 +49,16 @@ gen_hw_and_check_clean() {
 # Check generated files are up to date
 
 bad=0
+
+if [ "${OT_DESTRUCTIVE}" != "1" ]; then
+    cat >&2 <<EOM
+$0 aborting because OT_DESTRUCTIVE != 1.
+
+Warning: setting OT_DESTRUCTIVE=1 will enable the script to run 'git clean' and
+'git reset --hard', which may destroy your local work.
```

Comment:
It might be worth clarifying that this variable is intended to be set by the user rather than by another script. This makes it clear that running this script or another script that calls it like `ci/jobs/slow-lint.sh` requires manual intervention rather than there possibly being a bug.

```suggestion
Warning: this script runs 'git clean' and 'git reset --hard', which may destroy your local work.
Run this script with OT_DESTRUCTIVE=1 to enable the script to proceed.
```